tuf/signed/errors.go

```diff
@@ -15,11 +15,17 @@ type ErrInsufficientSignatures struct {
 
 func (e ErrInsufficientSignatures) Error() string {
 	candidates := strings.Join(e.MissingKeyIDs, ", ")
-	if e.FoundKeys == 0 {
-		return fmt.Sprintf("signing keys not available, need %d keys out of: %s", e.NeededKeys, candidates)
+	switch {
+	case len(e.MissingKeyIDs) < e.NeededKeys:
+		return fmt.Sprintf(
+			"cannot sign because while %d signatures are needed, an insufficient number of valid signing keys have been specified",
```

Comment:
this case could be broken down further into `len(e.MissingKeyIDS) + e.FoundKeys < e.NeededKeys` and `len(e.MissingKeyIDS) + e.FoundKeys >= e.NeededKeys` right?

For `len(e.MissingKeyIDS) + e.FoundKeys < e.NeededKeys`: role is broken
For `len(e.MissingKeyIDS) + e.FoundKeys >= e.NeededKeys`: @endophage's suggestion of `"insufficient signing keys available. %d signatures are required but only %d keys are available"`  This could be the base case in `default`